lib/FundDistribution/FundDistributionFields/FundDistributionFieldsFinalContainer.js

```diff
@@ -36,26 +39,51 @@ const FundDistributionFieldsFinalContainer = ({
   const [expenseClassesByFundId, setExpenseClassesByFundId] = useState({});
 
   useEffect(() => {
-    const expenseClassesFundsToFetch = fundDistribution?.filter(({ fundId }) => (
-      fundId && !expenseClassesByFundId[fundId]
-    ));
-
-    if (expenseClassesFundsToFetch?.length) {
-      expenseClassesFundsToFetch.forEach(({ fundId }) => fetchExpenseClasses(
-        fundId, expenseClassesByFundId, setExpenseClassesByFundId, mutator.fundExpenseClasses,
-      ));
+    const uniqueFundDistribution = uniqBy(fundDistribution, 'fundId');
```

Comment:
this change impacts orders

Reply:
From what I see, the old approach doesn't satisfy the current task due to expanse classes should be updated on the fiscal year id change.

Reply:
you still need to support other places

Reply:
@NikitaSedyx I have added a property `shouldRefetchExpenseClassesOnFiscalYearChange` to resolve concerns above